### PR TITLE
Problems

### PR DESCRIPTION
**Pull request by TiboldiRoberto**
Solutions for Problem 1, 2 and 3.
1. Problems solved in Visual Studio 2019.
2. Unit Tests - Used Google Test from Visual Studio 2019.

### CODE REVIEW

File: Problem1.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#include "pch.h"
+#include <gtest/gtest.h>
+
+using namespace std;
+
+int sumEvenNumbers(int[], int);
+int vec1[5] = { 2,13,15,6,2 };
+int vec2[10] = { 2,4,17,19,22,45,17,2,99,77 };
+
+TEST(Sum_Even_Number_Tests, Test1) {
```
Comment (RalucaMihaela) [inline Problem1.cpp]: Can you move the tests in separate files? Is better to have the tests in separate files from the actual implementation

**Comment by vladstanescu94** [comment]
Hello 👋, thanks for the solution. Can you leave some instructions on how to run the code and the tests?

File: Problem1.cpp (hunk 1 above)
Comment (vladstanescu94) [inline Problem1.cpp]: Can you think of more test cases here? What happens if the array is empty, or all numbers are odd?

File: Problem1.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	EXPECT_EQ(sumEvenNumbers(vec1, 5), 10);
+	EXPECT_EQ(sumEvenNumbers(vec2, 10), 31);
+}
+
+int sumEvenNumbers(int v[], int l) {
```
Comment (vladstanescu94) [inline Problem1.cpp]: Maybe the name of the params could be more descriptive.

File: Problem2.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+char missingCharacter(char c[])
+{
+	//Convert Vector
+	int l = strlen(c);
+	int* v;
+	v = new int[l];
+	for (int i = 0; i < l; i++)
+		v[i] = c[i];
+	//Compare Vector
```
Comment (vladstanescu94) [inline Problem2.cpp]: If you need two comments to describe what this function does, that means that the function is doing more than 1 thing. How can we fix this?

File: Problem3.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	EXPECT_EQ(lexicographicallyMinimalString("acaccczzxxrcv", "bcaccczzxxrca"), "abcacacccccczzxxrcazzxxrcv");
+}
+
+TEST(Lexicographically_Minimal_String, Test3) {
+	EXPECT_EQ(lexicographicallyMinimalString("adfrdsfacascfa", "aszczasr"), "aadfrdsfacascfaszczasr");
+}
+
+string lexicographicallyMinimalString(string j, string s) {
+	string result = "";
+	int l1 = j.length();
```
Comment (vladstanescu94) [inline Problem3.cpp]: Same as problem #1, I think it's better to use more descriptive names for the variables.

Comment (vladstanescu94) [reply Problem3.cpp]: It's hard to quickly read the code below because of the var names

**Review by vladstanescu94 — CHANGES_REQUESTED** [review]
Great solution 👍, some minor changes are needed.

File: Problem1.cpp (hunk 1 above)
Comment (TiboldiRoberto) [reply Problem1.cpp]: Of course, I will add new test cases

File: Problem1.cpp (hunk 2 above)
Comment (TiboldiRoberto) [reply Problem1.cpp]: I understand, I can change them.

File: Problem3.cpp (hunk 4 above)
Comment (TiboldiRoberto) [reply Problem3.cpp]: Ok, I can try to change variables names with descriptive names.

**Comment by TiboldiRoberto** [comment]
If you want to run the code in Visul Studio 2019, you need to:

1. Verify if the extension for Google Test is checked in Visual Studio Installer (Image 1) if not, add the extension.
2. Create a new project in VS with the template "Google Test" (Image 2)
3. Test Project Configuration: Default 
4. Copy or Add the source code in the test.cpp project
5. Run the test and hope it works well : )

VS-Installer-Check

VS-Create-GoogleTest-Project

File: Problem2.cpp (hunk 3 above)
Comment (TiboldiRoberto) [reply Problem2.cpp]: In this case we can use a single for and use cast to convert the characters to int so we don't need the vector of integers

File: Problem1.cpp [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+TEST(Sum_Even_Number_Tests, Test4) {
+	EXPECT_EQ(sumEvenNumbers(vector4, 5), 30);
+}
+
+TEST(Sum_Even_Number_Tests, Test5) {
+	EXPECT_EQ(sumEvenNumbers(vector5, 5), 0);
+}
+
+TEST(Sum_Even_Number_Tests, Test6) {
```
Comment (RalucaMihaela) [inline Problem1.cpp]: Where appropriate, can you change the test name into something that better defines what your test does?

**Review by RalucaMihaela — COMMENTED** [review]
Hello, great solution Roberto 🚀 I left some comments also for you to look over them.

File: Problem1.cpp (hunk 1 above)
Comment (TiboldiRoberto) [reply Problem1.cpp]: Ok, I can do it

File: Problem1.cpp (hunk 5 above)
Comment (TiboldiRoberto) [reply Problem1.cpp]: Yes, I can change the tests name
